src/resmom/mom_main.c

```diff
@@ -7330,7 +7330,7 @@ job_over_limit(job *pjob)
 	if ((pjob->ji_qs.ji_svrflags & JOB_SVFLG_HERE) == 0)	/* not MS */
 		return 0;
 
-	if (pjob->ji_nodekill != TM_ERROR_NODE) {
+	if ((pjob->ji_nodekill != TM_ERROR_NODE) && (pjob->ji_nodekill < pjob->ji_numnodes)) {
```

Comment:
Shouldn't we log a message if (pjob->ji_nodekill >= pjob->ji_numnodes)? Seems like it might be symptomatic of a deeper issue.

Reply:
Yes, we should log a warning message. I also figured out the culprit as to why pjob->ji_nodekill is getting to a node index that has already been released by this reliable job startup feature. I'll fix that too.